src/WebJobs.Script/Workers/Http/HttpScriptInvocationResultExtensions.cs

```diff
@@ -78,19 +79,29 @@ private static object GetBindingValue(DataType? bindingDataType, object outputBi
 
         internal static object GetHttpOutputBindingResponse(string bindingName, IDictionary<string, object> outputsFromWorker)
         {
-            HttpOutputBindingResponse httpOut = new HttpOutputBindingResponse();
+            dynamic httpOutput = new ExpandoObject();
+
             if (outputsFromWorker.TryGetValue(bindingName, out object outputBindingValue))
             {
                 try
                 {
-                    httpOut = JsonConvert.DeserializeObject<HttpOutputBindingResponse>(outputBindingValue.ToString());
+                    HttpOutputBindingResponse httpOutputBindingResponse = JsonConvert.DeserializeObject<HttpOutputBindingResponse>(outputBindingValue.ToString());
+
+                    httpOutput.StatusCode = httpOutputBindingResponse.StatusCode;
+                    httpOutput.Status = httpOutputBindingResponse.Status;
+                    httpOutput.Body = httpOutputBindingResponse.Body;
+                    httpOutput.Headers = httpOutputBindingResponse.Headers;
                 }
-                catch
+                catch (Exception e) when (e is JsonReaderException || e is JsonSerializationException)
                 {
-                    //ignore
+                    throw new HttpOutputDeserializationException("Output body should have 'StatusCode', 'Status', 'Body', and 'Headers'", e.Message);
```

Comment:
Instead of defining a new exception type - throw `InvalidOperationException` with inner exception as `e` ?